File: modules/07-scheduler/kotlin/app/Main.kt
fun main() {
    val total = 400_000_000L
    val w = Runtime.getRuntime().availableProcessors()
    val t0 = System.nanoTime()
    val acc = parallelSqrtSum(total, w)
    if (acc.isNaN()) println("nan")
    val ms = (System.nanoTime() - t0) / 1_000_000
    println("Kotlin sqrt-sum ${total / 1_000_000}M / $w threads: $ms ms")
}

File: modules/07-scheduler/ocaml/bin/main.ml
let () =
  let total = 400_000_000 in
  let workers = Domain.recommended_domain_count () in
  let t0 = Unix.gettimeofday () in
  let acc = Sched.parallel_sqrt_sum total workers in
  ignore (Sys.opaque_identity acc);
  let ms = int_of_float ((Unix.gettimeofday () -. t0) *. 1000.) in
  Printf.printf "OCaml  sqrt-sum %dM / %d domains: %d ms\n"
    (total / 1_000_000) workers ms
